### PR TITLE
Fix LongT5ForConditionalGeneration initialization of lm_head

### PR DESCRIPTION
**Pull request by eranhirs**
# What does this PR do?

LongT5ForConditionalGeneration does not work due to a missing initialization of its `lm_head`. Its weight matrices are initialized to zeros and then the logits are all zeros.

Code for reproduction:
```python
from transformers import AutoTokenizer, AutoModelForSeq2SeqLM

tokenizer = AutoTokenizer.from_pretrained("google/long-t5-local-base")
model = AutoModelForSeq2SeqLM.from_pretrained("google/long-t5-local-base")
model.generate(**tokenizer("Generate anything", return_tensors='pt'), max_length=200)
```

This PR contains code copied from [T5PreTrainedModel](https://github.com/huggingface/transformers/blob/v4.37.2/src/transformers/models/t5/modeling_t5.py#L830).

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
T5 has the same initialization scheme

**Comment by ArthurZucker** [comment]
Flax has a different init scheme we should be alright with cross PT FLax tests
